OpenSourcePolitics/decidim-module-half_sign_up
# fix: Modify SMS Behaviour

#### :tophat: Description
This PR fixes the behaviour of pasting directly from sms without doing the process of "copy paste" but by trusting our OS

#### :pushpin: Related Issues
- [Notion card](https://www.notion.so/opensourcepolitics/Am-liorations-half-sign-up-c40077a5469a495d80bebbc264ed8193?pvs=4)

⚠️ THIS REQUIRES A BIT OF SETUP PLEASE READ THE PROCESS OF TESTING

SETUP:
* Generate your dev_app `bundle exec rake development_app`
* In your dev_app create a file `config/initializers/half_signup.rb` and fill it with 
```ruby
# config/initializers/half_singup.rb
Decidim::HalfSignup.configure do |config|
  config.show_tos_page_after_signup = true
  # change this to false, if you don't want to redirect the user to the tos agreement page

  config.auth_code_length = 4
  # change this to other values if you want to change the length of generated code (be advised to remain in an acceptable limits for the sake of best performance)

  config.default_countries = [:us]
  # change ':us' to the country/countries you want to be shown at the top(the first option will be selected by default).
end
```
* Search for your IP Address by typing in your terminal `ifconfig | grep netmask | grep -v 127.0.0.1 | awk '{print $2}'`
* When it generated, launch for a first time your app and go to the system
* Modify the generated organization by changing the host and secondary host by your previously owned ip address
* Leave the app
* Restart it using `bundle exec rails start -b (your_ip_address)`
* Using your computer connect to (your_ip_address):3000
* Connect to admin@example.org
* Go to backoffice
* Access to settings then authentication settings
* Enable half-signup
* Using your smartphone connect to (your_ip_address):3000
* NOW You're set for the test

TEST:
* Try to signup
* Go to SMS
* Fill in a smartphone number
* Copy the code that is printed on the screen
* Open your text box
* Click on the "paste" icon (acts the same as if it was from your messages)
* Check that it works correctly

#### Tasks
- [ ] Change JS Behaviour
- [ ] Change input from text to number
- [ ] Make sure that you cannot put multiple characters in the same box


### :camera: Screenshots
*Please add screenshots of the changes you're proposing if related to the UI*
